modules/couchbase/collect.go

```diff
@@ -0,0 +1,164 @@
+package couchbase
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/netdata/go.d.plugin/agent/module"
+	"github.com/netdata/go.d.plugin/pkg/web"
+)
+
+const (
+	urlPathBucketsStats = "/pools/default/buckets"
+)
+
+func (cb *Couchbase) collect() (map[string]int64, error) {
+
+	ms := cb.scrapeCouchbase()
+	if ms.empty() {
+		return nil, nil
+	}
+	collected := make(map[string]int64)
+	err := cb.collectBasicStats(collected, ms)
+	if err != nil {
+		return nil, fmt.Errorf("error on creating a connection: %v", err)
+	}
+
+	return collected, nil
+}
+
+func (cb Couchbase) collectBasicStats(collected map[string]int64, ms *cbMetrics) error {
+	for _, b := range ms.BucketsStats {
+
+		if !cb.collectedBuckets[b.Name] {
+			cb.collectedBuckets[b.Name] = true
+			cb.addBucketToCharts(b.Name)
+		}
+
+		bs := b.BasicStats
+		collected[indexDimID(b.Name, "quota_used")] = int64(bs.QuotaPercentUsed)
+		collected[indexDimID(b.Name, "ops")] = int64(bs.OpsPerSec)
+		collected[indexDimID(b.Name, "fetches")] = int64(bs.DiskFetches)
+		collected[indexDimID(b.Name, "item_count")] = int64(bs.ItemCount)
+		collected[indexDimID(b.Name, "disk")] = int64(bs.DiskUsed)
+		collected[indexDimID(b.Name, "data")] = int64(bs.DataUsed)
+		collected[indexDimID(b.Name, "mem")] = int64(bs.MemUsed)
+		collected[indexDimID(b.Name, "num_non_resident")] = int64(bs.VbActiveNumNonResident)
+	}
+
+	return nil
+}
+
+func (cb *Couchbase) addBucketToCharts(bucket string) error {
```

Comment:
```suggestion
func (cb *Couchbase) addBucketToCharts(bucket string) {
```

no need in returning an error